lib/apartment/adapters/postgresql_adapter.rb

```diff
@@ -185,6 +185,7 @@ def patch_search_path(sql)
           .select {|line| check_input_against_regexps(line, PSQL_DUMP_BLACKLISTED_STATEMENTS).empty?}
           .prepend(search_path)
           .join("\n")
+          .gsub(/#{default_tenant}\./, "#{current}.")
```

Comment:
This is also namespacing database functions (specifically `uuid_generate_v4()`), causing them to fail because they are in `public`. It might be best to simply `.gsub(/#{default_tenant}\./, "")`, which I believe moves this back to parity with the original implementation.

Reply:
I ended up here:
```ruby
      def patch_search_path(sql)
        search_path = "SET search_path = \"#{current}\", #{default_tenant};"

        sql
          .split("\n")
          .select {|line| check_input_against_regexps(line, PSQL_DUMP_BLACKLISTED_STATEMENTS).empty?}
          .map do |line|
            if [/default/i, /uuid_generate/i, /geography/i].any? { |r| line =~ r }
              line
            else
              line.gsub("#{default_schema}\.", "#{current}.")
            end
          end .prepend(search_path)
          .join("\n")
      end
```

The `geography` is in there because of PostGIS, which lives in the public schema. 

I also had to change `with_pg_env { `pg_dump -a --inserts -t schema_migrations  -n #{default_tenant} #{dbname}` }` to not dump the `ar_internal_metadata` table, because I was getting a duplicate error on the primary key.

Reply:
Ok - actually, I had to remove `default` from the regex list, because it messed up the primary key assigment. So it looks like maybe we'll need to allow a list of excluded postgresql functions/types/etc to be put on the config object that won't be un-namespaced.

I'm going to give it a shot in my own branch.

Reply:
@shterrett can you give me a sample of your db_dump which cause this ?

Because i'd `uuid_generate_v4` and `ar_internal_metadata` in my dump, then i didn't have the problem.

Reply:
Yeah - I think my other problem is I'm on postgres 10.3

Here's a specific table - the original:
```sql
CREATE TABLE public.zones (
    id integer NOT NULL,
    created_at timestamp without time zone NOT NULL,
    updated_at timestamp without time zone NOT NULL,
    time_zone character varying NOT NULL,
    uuid uuid DEFAULT public.uuid_generate_v4(),
    abbreviation character varying,
    name character varying,
    point public.geography(Point,4326),
    country_id integer NOT NULL,
    trip_stop_notification_horizon integer DEFAULT 15 NOT NULL,
    trip_type public.trip_type_enum DEFAULT 'full'::public.trip_type_enum,
    transfer_window_minutes integer DEFAULT 20 NOT NULL,
    operator_id integer
);
```

And after the gsub:
```sql
```